sqlglot/dialects/tsql.py

```diff
@@ -524,6 +524,13 @@ class Parser(parser.Parser):
             "TIMEFROMPARTS": _build_timefromparts,
         }
 
+        RANGE_PARSERS = {
+            **parser.Parser.RANGE_PARSERS,
+            TokenType.DCOLON: lambda self, this: self.expression(
+                exp.ScopeResolution, this=this, expression=self._parse_function()
```

Comment:
Let's use a more lenient parser for the RHS, eg. the 2nd [link](https://learn.microsoft.com/en-us/sql/t-sql/statements/grant-server-principal-permissions-transact-sql?view=sql-server-ver16) shows this syntax:

```sql
GRANT VIEW DEFINITION ON LOGIN::EricKurjan TO RMeyyappan   
    WITH GRANT OPTION
```